rmc/utils/constraint.py

```diff
@@ -1831,6 +1834,98 @@ def format_rmc_browser_ht(freeze: int, overwrite_temp: bool) -> None:
     ht.write(rmc_browser.versions[freeze].path, overwrite=True)
 
 
+def create_rmc_release_downloads(
+    freeze: int,
+    overwrite: bool,
+    gencode_version: str = GENCODE_VERSION,
+    vep_version: str = VEP_VERSION,
+) -> None:
+    """
+    Create downloadable files to be publicly released on the gnomAD browser.
+
+    Function adds gene IDs to RMC HT and exports RMC results to TSVs for release.
+
+    The two TSV types are:
+    - TSV with transcripts that had evidence of RMC
+    - TSV listing all transcripts that were searched for but did not have evidence of RMC
+
+    .. note::
+        - Function assumes the global annotation `transcripts_no_rmc` exists
+        in `rmc_browser.ht()`
+        - Function assumes the following fields exist in `regions` struct on `rmc_browser.ht()`:
+            - start_coordinate
+            - stop_coordinate
+            - start_aa
+            - stop_aa
+            - obs
+            - exp
+            - oe
+            - chisq
+            - p
+        - Function assumes the following fields exist on the gene constraint HT:
+            - obs_mis
+            - exp_mis
+            - oe_mis
+            - mis_z
+
+    :param freeze: RMC data freeze number.
+    :param overwrite: Whether to overwrite RMC download HT.
+    :param gencode_version: Version of GENCODE used to annotate variants.
+        Default is `GENCODE_VERSION`.
+    :param vep_version: Version of VEP used to annotate variants.
+        Default is `VEP_VERSION`.
+    :return: None; writes TSVs to resource paths.
+    """
+    logger.info("Preparing browser reformatted HT for release...")
+    ht = rmc_browser.versions[freeze].ht()
+
+    # Add gene IDs from `transcript_ref` resource
+    transcript_ref_ht = transcript_ref.ht()
+    ht = ht.annotate(
+        gene_name=transcript_ref_ht[ht.transcript].gnomad_gene,
+        gene_id=transcript_ref_ht[ht.transcript].gencode_gene_id,
+    )
+    # Add GENCODE and VEP versions to globals
+    ht = ht.annotate_globals(gencode_version=gencode_version, vep_version=vep_version)
+    ht = ht.checkpoint(
+        rmc_downloads_resource_paths(get_ht_path=True, freeze=freeze),
+        overwrite=overwrite,
+        _read_if_exists=not overwrite,
+    )
+
+    # Get transcripts that were searched for but had no evidence of RMC
+    transcripts_no_rmc = ht.transcripts_no_rmc
+    # Filter gene constraint results to keep only trascripts without evidence of RMC
+    gene_constraint_ht = constraint_ht.ht().select_globals().key_by("transcript")
+    gene_constraint_ht = gene_constraint_ht.filter(
```

Comment:
Do you want to add GENCODE gene IDs to this table too since we add it to the other one?

Reply:
good idea!